recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+                self._cmake.definitions["CAF_NO_OPENSSL"] = not self.options.with_openssl
+                for define in ["CAF_NO_EXAMPLES", "CAF_NO_TOOLS", "CAF_NO_UNIT_TESTS", "CAF_NO_PYTHON"]:
+                    self._cmake.definitions[define] = "ON"
+                self._cmake.definitions["CAF_BUILD_STATIC"] = not self.options.shared
+                self._cmake.definitions["CAF_BUILD_STATIC_ONLY"] = not self.options.shared
+            else:
+                self._cmake.definitions["CAF_ENABLE_OPENSSL_MODULE"] = self.options.with_openssl
+                for define in ["CAF_ENABLE_EXAMPLES", "CAF_ENABLE_TOOLS", "CAF_ENABLE_TESTING"]:
+                    self._cmake.definitions[define] = "OFF"
+            self._cmake.definitions["CAF_LOG_LEVEL"] = self.options.log_level.value.upper()
+            if self.settings.compiler == 'clang':
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread -ldl"
+            else:
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread"
+                if self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5.0":
+                    self._cmake.definitions["CMAKE_SHARED_LINKER_FLAGS"] = "-pthread"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._cmake_configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
+        cmake = self._cmake_configure()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        suffix = "_static" if not self.options.shared and Version(self.version) <= "0.17.6" else ""
+
+        self.cpp_info.names["cmake_find_package"] = "CAF"
+        self.cpp_info.names["cmake_find_package_multi"] = "CAF"
+
+        self.cpp_info.components["core"].names["pkg_config"] = "caf_core{}".format(suffix)
+        self.cpp_info.components["core"].libs = ["caf_core{}".format(suffix)]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["core"].system_libs = ["pthread", "m"]
+
+        self.cpp_info.components["io"].names["pkg_config"] = "caf_io{}".format(suffix)
+        self.cpp_info.components["io"].libs = ["caf_io{}".format(suffix)]
+        self.cpp_info.components["io"].requires = ["core"]
+        if self.settings.os == "Windows":
+            self.cpp_info.components["io"].system_libs = ["ws2_32", "iphlpapi", "psapi"]
+
+        if self.options.with_openssl:
+            self.cpp_info.components["openssl"].names["pkg_config"] = f"caf_openssl{suffix}"
+            self.cpp_info.components["openssl"].libs = [f"caf_openssl{suffix}"]
+            self.cpp_info.components["openssl"].requires = ["io", "openssl::openssl"]
```

Comment:
```suggestion
        self.cpp_info.components["caf_core"].names["cmake_find_package"] = "core"
        self.cpp_info.components["caf_core"].names["cmake_find_package_multi"] = "core"
        self.cpp_info.components["caf_core"].libs = ["caf_core{}".format(suffix)]
        if self.settings.os == "Windows":
            self.cpp_info.components["caf_core"].system_libs = ["iphlpapi"]
        elif self.settings.os == "Linux":
            self.cpp_info.components["caf_core"].system_libs = ["pthread", "m"]

        self.cpp_info.components["caf_io"].names["cmake_find_package"] = "io"
        self.cpp_info.components["caf_io"].names["cmake_find_package_multi"] = "io"
        self.cpp_info.components["caf_io"].libs = ["caf_io{}".format(suffix)]
        self.cpp_info.components["caf_io"].requires = ["caf_core"]
        if self.settings.os == "Windows":
            self.cpp_info.components["caf_io"].system_libs = ["ws2_32"]

        if self.options.with_openssl:
            self.cpp_info.components["caf_openssl"].names["cmake_find_package"] = "openssl"
            self.cpp_info.components["caf_openssl"].names["cmake_find_package_multi"] = "openssl"
            self.cpp_info.components["caf_openssl"].libs = ["caf_openssl{}".format(suffix)]
            self.cpp_info.components["caf_openssl"].requires = ["caf_io", "openssl::openssl"]
```

Reply:
I don't see official pkg-config names.
iphlpapi is a dependency of CAF::core on Windows
I don't see psapi in dependencies on Windows

Reply:
Here is the content of CAFTargets.cmake in 0.18.0 if static on Windows, to help with components:

```cmake
...
# Create imported target CAF::internal
add_library(CAF::internal INTERFACE IMPORTED)

# Create imported target CAF::test
add_library(CAF::test INTERFACE IMPORTED)

# Create imported target CAF::core
add_library(CAF::core STATIC IMPORTED)

set_target_properties(CAF::core PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Threads::Threads;iphlpapi;\$<LINK_ONLY:CAF::internal>"
)

# Create imported target CAF::io
add_library(CAF::io STATIC IMPORTED)

set_target_properties(CAF::io PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "CAF::core;\$<\$<CXX_COMPILER_ID:MSVC>:ws2_32>;\$<LINK_ONLY:CAF::internal>"
)

# Create imported target CAF::openssl
add_library(CAF::openssl STATIC IMPORTED)

set_target_properties(CAF::openssl PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "CAF::io;OpenSSL::SSL;OpenSSL::Crypto;\$<LINK_ONLY:CAF::internal>"
)
...
```

Reply:
We could but is it needed?  The test_package already links ok with
```
target_link_libraries(${PROJECT_NAME} CAF::core CAF::io CAF::openssl)
```

Reply:
There is no reason to add useless dependencies, and test_package works because you link all targets. Maybe it fails if you link only CAF::core

Moreover, too generic conan components names should be avoided when pkg_config names are not overridden.

Reply:
I removed the pkg_config items and moved iphlapi to core.  

I was wondering why the `self.cpp_info.components["xxx"].names["yyy"] = "zzz"` were needed.

Reply:
As I said, we must avoid too generic conan components names because it might lead to side effects in pkg_config generator. This is why I've add `caf_` prefix to all components in my review, and therefore `self.cpp_info.components["xxx"].names["yyy"] = "zzz"` is needed.

Reply:
Changed